src/node/blockstorage.cpp

```diff
@@ -348,6 +331,31 @@ void BlockManager::Unload()
     }
 
     m_block_index.clear();
+
+    m_blockfile_info.clear();
+    m_last_blockfile = 0;
+    m_dirty_blockindex.clear();
+    m_dirty_fileinfo.clear();
+}
+
+bool BlockManager::WriteBlockIndexDB()
+{
```

Comment:
fa467f3913918701 Missing `AssertLockHeld(::cs_main)` in the definition of this new helper. Added in #24002.